packages/aws-cdk-lib/core/lib/cfn-mapping.ts

```diff
@@ -63,30 +63,41 @@ export class CfnMapping extends CfnRefElement {
   }
 
   /**
-   * @returns A reference to a value in the map based on the two keys.
+   * @returns A reference to a value in the map based on the two keys. If mapping is lazy, the value from the map or default value is returned instead of the reference.
```

Comment:
I'm pretty sure "lazy" means: only render the actual mapping to the template if there is at least one value that actually needs to be looked up from it at deploy time.

Conversely: if no value is ever read from the map, or all values can be resolved at synth time, then there is no need to render the map.

The line of documentation you added here doesn't help me understand that :).

Reply:
Felt that it wasn't accurate to say it returns a reference is sometimes it just passes a value. Also could help debug if the short circuit was not documented. Can add a line about the mapping behavior too